groovy/gradle/src/org/netbeans/modules/gradle/api/execute/GradleCommandLine.java

```diff
@@ -825,45 +826,56 @@ public void setStackTrace(StackTrace st) {
         }
     }
 
-    private void addGradleSettingJvmargs(File projectDir, List<String> jvmargs) {
+    final static void addGradleSettingJvmargs(File rootDir, List<String> jvmargs) {
         List<File> propFiles = new ArrayList<>();
+        propFiles.add(new File(GradleSettings.getDefault().getGradleUserHome(), GradleFiles.GRADLE_PROPERTIES_NAME));
 
-        if (projectDir == null) {
-            File gradleHome = GradleSettings.getDefault().getGradleUserHome();
-            File f = new File(gradleHome, GradleFiles.GRADLE_PROPERTIES_NAME);
-            if (f.exists()) {
-                propFiles.add(f);
-            }
-        } else {
-            propFiles.addAll(new GradleFiles(projectDir).getPropertyFiles());
+        if (rootDir != null) {
+            propFiles.addAll(new GradleFiles(rootDir).getPropertyFiles());
         }
-
+        //TODO: Theoretically the Gradle Distribution dir can have a gradle.properties
+        //      however computing that is not really easy, at the moment we do
+        //      not support that one.
         for (File f : propFiles) {
-            try (InputStream in = new FileInputStream(f)) {
-                Properties props = new Properties();
-                props.load(in);
-                if (props.containsKey(PROP_JVMARGS)) {
-                    jvmargs.addAll(Arrays.asList(props.getProperty(PROP_JVMARGS).split("\\s+"))); //NOI18N
+            if (f.canRead()) {
+                try (InputStream in = new FileInputStream(f)) {
+                    Properties props = new Properties();
+                    props.load(in);
+                    if (props.containsKey(PROP_JVMARGS)) {
+                        String[] args = Utilities.parseParameters(props.getProperty(PROP_JVMARGS));
+                        for (String arg : args) {
+                            jvmargs.add(removeSingleQuotes(arg));
+                        }
+                        break;
+                    }
+                } catch (IOException ex) {
+                    LOGGER.log(Level.INFO, "Cannot read property file: '" + f.getAbsolutePath() + "' as: " + ex.getMessage());
                 }
-            } catch (IOException ex) {
-                LOGGER.log(Level.INFO, "Cannot read property file: '" + f.getAbsolutePath() + "' as: " + ex.getMessage());
             }
         }
     }
 
+    final static String removeSingleQuotes(String s) {
```

Comment:
What is the reason for the special handling of single quotes after they are already parsed?

Reply:
Unfortunately the command line parser in NetBeans does not parse the single quotes. It's maybe a bug there. It was easier to add a workaround to the most common case here than fix it there. Also the impact of fixing it in the utilities might have unwanted consequences.

Reply:
But this only works for the simplest of cases doesn't it? This:

```java
import java.util.Arrays;

class test {
	public static void main(String[] argv) {
		System.out.println(Arrays.toString(argv));
	}
}
```

Run as:

```
matthias@athena:~$ java -cp . test -Dsomething="space value"
[-Dsomething=space value]
matthias@athena:~$ java -cp . test -Dsomething='space value'
[-Dsomething=space value]
matthias@athena:~$ 
```

shows, that also embedded single quotes need to be removed. Is this really needed or just a theoretical issue?

Reply:
Well thought it over and fixed the BaseUtilities.parseParameters instead in #1829 after all it was fun.